istioctl/cmd/kubeuninject.go

```diff
@@ -197,23 +197,10 @@ func removeDNSConfig(podDNSConfig *corev1.PodDNSConfig) {
 	if podDNSConfig == nil {
 		return
 	}
-
-	l := len(podDNSConfig.Searches)
-	index := 0
-	for index < l {
-		s := podDNSConfig.Searches[index]
-		if strings.Contains(s, "global") {
-			if index < len(podDNSConfig.Searches)-1 {
-				podDNSConfig.Searches = append(podDNSConfig.Searches[:index],
-					podDNSConfig.Searches[index+1:]...)
-			} else {
-				podDNSConfig.Searches = podDNSConfig.Searches[:index]
-			}
-			// reset to 0
-			index = 0
-			l = len(podDNSConfig.Searches)
-		} else {
-			index++
+	for i := 0; i < len(podDNSConfig.Searches); i++ {
```

Comment:
Please run `make gen` to include any changed files

Reply:
The code LGTM

Reply:
Use `make gen` , `git diff.` found that nothing has changed
Finally use git apply <(curl -sL "https://gcsweb.istio.io/gcs/istio-prow/pr-logs/pull/istio_istio/43901/gencheck_istio/1635356338655596544/artifacts/check-clean-repo-diff.patch")

Reply:
This is strange. The code you've written should not affect protoc, maybe try to add the code to the latest commit. I haven't encountered this issue today.

Reply:
I tried to switch to the master branch using `git fetch --all && git rebase upstream/master && git switch feature/OptimizationremoveDNSConfig && git rebase upstream/master`, but nothing changes, and I tried `git push -f`